### ISSUE TITLE
does this also work for a server with a private ip in a intranet?

### ISSUE THREAD

**Issue by Freundschaft** [chaifeng/ufw-docker#67]
I see that this solution works for a server with a public address, but how does it work if the server also has a private ip address lets say on a company intranet?
ive tried the script and it doesnt work and im wondering if its connected to the overlapping private ip ranges

**Comment by Suertzz** [comment]
Yes, same issue here did u find any solution @Freundschaft ? Im on private range and its certainly overlapping the docker ranges

**Comment by Freundschaft** [comment]
to be honest i just decided to not use ufw on the host that runs docker and configure firewall rules on the network layer firewall instead on my cloud service, unfortunately i didnt find a solution for this

**Comment by chaifeng** [comment]
You could modify the file `/etc/ufw/after.rules` to match your company's internal network.


	-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 192.168.0.0/16
	-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 10.0.0.0/8
	-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 172.16.0.0/12
	-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 192.168.0.0/16
	-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 10.0.0.0/8
	-A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 172.16.0.0/12

In the above code block there three network addresses match all private network addresses
```
10.0.0.0/8
172.16.0.0/12
192.168.0.0/16
```

Replace the above code block with your own:

Each such the following two lines as a group, and only the last parameter needs to be changed.
```
-A DOCKER-USER -j DROP -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 192.168.1.0/24
-A DOCKER-USER -j DROP -p udp -m udp --dport 0:32767 -d 192.168.1.0/24
```

You could add multiple lines of such code groups.

**Comment by chaifeng** [comment]
> You could modify the file `/etc/ufw/after.rules` to match your company's internal network.
> 
> ```
> -A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 192.168.0.0/16
> -A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 10.0.0.0/8
> -A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 172.16.0.0/12
> -A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 192.168.0.0/16
> -A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 10.0.0.0/8
> -A DOCKER-USER -j ufw-docker-logging-deny -p udp -m udp --dport 0:32767 -d 172.16.0.0/12
> ```
> 
> In the above code block there three network addresses match all private network addresses
> 
> ```
> 10.0.0.0/8
> 172.16.0.0/12
> 192.168.0.0/16
> ```
> 
> Replace the above code block with your own:
> 
> Each such the following two lines as a group, and only the last parameter needs to be changed.
> 
> ```
> -A DOCKER-USER -j DROP -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 192.168.1.0/24
> -A DOCKER-USER -j DROP -p udp -m udp --dport 0:32767 -d 192.168.1.0/24
> ```
> 
> You could add multiple lines of such code groups.

Sorry for my incorrect reply.

I forgot the following three lines, which allow all private network access to docker services.

    -A DOCKER-USER -j RETURN -s 10.0.0.0/8
    -A DOCKER-USER -j RETURN -s 172.16.0.0/12
    -A DOCKER-USER -j RETURN -s 192.168.0.0/16

you also need to change these three lines to match your docker network and trusted internal network

**Comment by archont00** [comment]
I stumbled upon a similar problem: container A (SRC=172.17.30.4) trying to access container B (`ldap` 172.17.8.3) via host private address (10.11.13.21) and port mapped to container B.

```
# ufw-docker allow ldap
allow ldap 389/tcp ldap_default
ufw route allow proto tcp from any to 172.17.8.3 port 389 comment allow ldap 389/tcp ldap_default
Rule added
allow ldap 636/tcp ldap_default
ufw route allow proto tcp from any to 172.17.8.3 port 636 comment allow ldap 636/tcp ldap_default
Rule added
```

```
[UFW BLOCK] IN=br-219f8c918d47 OUT= PHYSIN=veth94971af MAC=02:42:3c:a3:35:bc:02:42:ac:11:1e:04:08:00 SRC=172.17.30.4 DST=10.11.13.21 LEN=60 TOS=0x00 PREC=0x00 TTL=64 ID=62339 DF PROTO=TCP SPT=36418 DPT=389 WINDOW=64240 RES=0x00 SYN URGP=0
```
It is dropped because the destination is a private address and the source is not a private address:
```
-A DOCKER-USER -j RETURN -s 172.16.0.0/12
-A DOCKER-USER -j ufw-docker-logging-deny -p tcp -m tcp --tcp-flags FIN,SYN,RST,ACK SYN -d 10.0.0.0/8
```

Shouldn't `ufw-docker allow container` allow not only connections from public networks but also from other containers on the same host routed via host's private ip?

After-thought: Eh, this probably does not have a good solution other than adding the docker networks (172.17.0.0/16 in my case) to the list of private networks and allow every container to access to host private ip.
